zerver/lib/upload.py

```diff
@@ -356,7 +356,7 @@ def ensure_medium_avatar_image(self, user_profile: UserProfile) -> None:
         conn = S3Connection(settings.S3_KEY, settings.S3_SECRET_KEY)
         bucket = get_bucket(conn, bucket_name)
         key = bucket.get_key(file_path)
-        image_data = force_bytes(key.get_contents_as_string())
+        image_data = key.get_contents_as_string()
```

Comment:
@showell @gnprice, here, `key.get_contents_as_string()` return the type bytes. So when I check the type of `image_data` I get `<class 'bytes'>`. However I get the mypy error `zerver/lib/upload.py:361: error: Argument 1 to "resize_avatar" has incompatible type "str"; expected "bytes"`. I have test of this function in the previous commit.

Reply:
http://boto.readthedocs.io/en/latest/ref/s3.html Here we can see that in the function `get_contents_as_string()`, The default encoding is `None` and it should return `bytes`. But I am stuck at the mypy error.

Reply:
The problem here is that `resize_avatar` expects `bytes` for the first argument.  If the function `get_contents_as_string()` returns `bytes` then it's kind of a bad name for the function, but that's boto's fault, not ours.  Also, their stubs are apparently broken.  You can use `type: ignore` here.

~~~
-        resized_medium = resize_avatar(image_data, MEDIUM_AVATAR_SIZE)
+        # image_data is actually `bytes` but the boto stubs are wrong
+        resized_medium = resize_avatar(image_data, MEDIUM_AVATAR_SIZE)  # type: ignore